### PR TITLE
Port most snapshot/restore static bwc tests to qa:full-cluster-restart

### PR DESCRIPTION
**Pull request by nik9000**
Ports all of RepositoryUpgradabilityIT to qa:full-cluster-restart and ports as much of RestoreBackwardsCompatIT as possible into qa:full-cluster-restart.

### CODE REVIEW

**Comment by nik9000** [comment]
Still having to do some cleanup for pre-5.5 indices.

**Comment by nik9000** [comment]
Ok. I fixed this for pre-5.5 versions.

File: qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java [hunk 1]
```diff
@@ -612,58 +680,110 @@ public void testSnapshotRestore() throws IOException {
...
+
+        // Clean up the index for the next iteration
+        client().performRequest("DELETE", "/restored_*");
+
+        // Check settings added by the restore process
+        map = toMap(client().performRequest("GET", "/_cluster/settings", singletonMap("flat_settings", "true")));
+        Map<String, Object> expected = new HashMap<>();
+        expected.put("transient", emptyMap());
+        expected.put("persistent", singletonMap("cluster.routing.allocation.exclude.test_attr", oldClusterVersion.toString()));
+        if (false == expected.equals(map)) {
```
Comment (martijnvg) [inline qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java]: maybe reverse this check? (`expected.equals(map) == false`)

**Review by martijnvg — APPROVED** [review]
LGTM. Left one nit.

**Comment by nik9000** [comment]
Thanks for reviewing @martijnvg!

**Pull request merged**